hetmech/degree_weight.py

```diff
@@ -125,18 +129,23 @@ def dwpc_duplicated_metanode(graph, metapath, duplicate=None, damping=0.5):
     dwpc_matrix = None
     row_names = None
     for metaedge in metapath:
-        rows, cols, adj_mat = metaedge_to_adjacency_matrix(graph, metaedge)
+        rows, cols, adj_mat = \
+            metaedge_to_adjacency_matrix(graph, metaedge, auto=auto,
+                                         matrix_type=mat_type)
         adj_mat = dwwc_step(adj_mat, damping, damping)
         if dwpc_matrix is None:
             row_names = rows
             dwpc_matrix = adj_mat
         else:
             dwpc_matrix = dwpc_matrix @ adj_mat
+            if auto and sparse.issparse(dwpc_matrix):
+                if dwpc_matrix.nnz > (1/3) * numpy.prod(dwpc_matrix.shape):
+                    dwpc_matrix = dwpc_matrix.toarray()
         if metaedge.target == duplicate:
-            # # scipy.sparse method threw ValueError:
-            # # Different number of diagonals and offsets.
-            # diag_matrix = scipy.sparse.diags(dwpc_matrix.diagonal())
-            diag_matrix = numpy.diag(dwpc_matrix.diagonal())
+            if sparse.issparse(dwpc_matrix):
```

Comment:
Just noticed that the above added method only works for square matrices. I will add functionality for non-square matrices.